### PR TITLE
bugfix: S3C-4533 prevent suspending version on locked buckets

### PR DESCRIPTION
**Pull request by ilkescality**

### CODE REVIEW

File: tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js [hunk 1]
```diff
@@ -146,11 +145,40 @@ describe('aws-node-sdk test bucket versioning', function testSuite() {
...
-        // s3.getBucketVersioning(params, done);
         s3.getBucketVersioning(params, (error, data) => {
             assert.strictEqual(error, null);
             assert.deepStrictEqual(data, { Status: 'Enabled' });
             done();
         });
     });
 });
+
+describe.only('aws-node-sdk test bucket versioning with object lock', () => {
```
Comment (rahulreddy) [inline tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js]: @ilkescality You may have forgotten `.only` here

Comment (ilkescality) [reply tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js]: ah!! just removed it, thanks!

File: tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js [hunk 2]
```diff
@@ -146,11 +145,40 @@ describe('aws-node-sdk test bucket versioning', function testSuite() {
...
+        s3.createBucket({
+            Bucket: bucket,
+            ObjectLockEnabledForBucket: true,
+        }, done);
+    });
+
+    // delete bucket after testing
+    after(done => s3.deleteBucket({ Bucket: bucket }, done));
+
+    it('should not accept empty versioning configuration', done => {
```
Comment (jonathan-gramain) [inline tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js]: the test description might be better as "should not accept versioning configurations"

**Review by jonathan-gramain — APPROVED** [review]

File: tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js (hunk 2 above)
Comment (ilkescality) [reply tests/functional/aws-node-sdk/test/bucket/testBucketVersioning.js]: looks like I forgot the description from above test, just updated.

**Review by anurag4DSB — APPROVED** [review]

**Pull request merged**